### PR TITLE
kernel: change PADTO check to allow up to 4x padding

### PR DESCRIPTION
**Pull request by flammit**
also optionally remove PADTO from the search action space with BEAM_PADTO=0.

for tiny red / `HSA=1`:

| branch  | settings | setup (m:ss) | step time (ms) | flops (TF) |
| --- | --- | --- | --- |  --- |
| master | `SPLIT_REDUCEOP=1`| 16:25 | 420ms | 90TF |
| kernel_loosen_pad_limit | `SPLIT_REDUCEOP=1` | 16:13 | 372ms | 102TF |
| kernel_loosen_pad_limit | `SPLIT_REDUCEOP=1` + **kitchen_sink | 35:20 | 340ms | 112TF |

for tiny green / `CUDA=1`:

| branch  | settings | setup (m:ss) | step time (ms) | flops (TF) |
| --- | --- | --- | --- |  --- |
| master | `SPLIT_REDUCEOP=1`| 20:15 | 300ms | 127TF |
| kernel_loosen_pad_limit | `SPLIT_REDUCEOP=1` | 20:04 | 264ms | 143TF |
| kernel_loosen_pad_limit | `SPLIT_REDUCEOP=1` + **kitchen_sink | 40:58 | 260ms | 146TF |

** kitchen sink: `TRAIN_BEAM=6 IGNORE_JIT_FIRST_BEAM=1 BEAM_UOPS_MAX=1500 BEAM_UPCAST_MAX=128 BEAM_LOCAL_MAX=1024 BEAM_MIN_PROGRESS=10 BEAM_PADTO=0`

### CODE REVIEW

**Comment by chenyuxyz** [comment]
note: hip tests took 17 minutes

**Comment by chenyuxyz** [comment]
oh rhip slow is not related to this pr, it was slow on master too

**Comment by flammit** [comment]
> oh rhip slow is not related to this pr, it was slow on master too

Yup - my bad on the previous TC reduce choice PR :/

**Comment by chenyuxyz** [comment]
i think it's fixed now. let's get this merge and update the setup

**Comment by flammit** [comment]
Will do - just finishing the tests now

File: tinygrad/features/search.py [hunk 1]
```diff
@@ -85,8 +85,8 @@ def bufs_from_lin(lin:Linearizer, allocate:bool=True) -> List[Buffer]:
 def get_linearizer_actions(lin:Linearizer, include_0=True) -> Dict[int, Linearizer]:
   acted_lins, max_up, max_lcl = {0:lin} if include_0 else {}, getenv("BEAM_UPCAST_MAX", 256), getenv("BEAM_LOCAL_MAX", 256)
   for i,a in enumerate(actions):
-    if a.axis is not None and a.axis >= lin.shape_len: continue
-    if a.axis is not None and lin.full_shape[a.axis] == a.amt and Opt(a.op, a.axis, 0) in actions: continue
+    if a.axis is not None and a.op is not OptOps.TC and a.axis >= lin.shape_len: continue
+    if a.axis is not None and a.op is not OptOps.TC and lin.full_shape[a.axis] == a.amt and Opt(a.op, a.axis, 0) in actions: continue
```
Comment (chenyuxyz) [inline tinygrad/features/search.py:89]: oh i meant
```
if a.axis is not None and a.op is not OptOps.TC:
  if (a.axis >= lin.shape_len) or (lin.full_shape[a.axis] == a.amt and Opt(a.op, a.axis, 0) in actions): continue
```

if another pr. see if you want to squeeze it here when you rebase

Comment (flammit) [reply tinygrad/features/search.py]: ohhhhh sorry hahahaha.  I liked this way too.

**Pull request merged**